docs/mkdocstrings_helper.py

```diff
@@ -0,0 +1,205 @@
+
+    Args:
+        markdown_str (str): Markdown string with line breaks to write to file.
+        filename (str): Filename without the .md
+        out_path (str): The output directory
+    """
+    if not markdown_str:
+        # Don't write empty files
+        return
+
+    md_file = filename
+    if not filename.endswith(".md"):
+        md_file = filename + ".md"
+
+    print(f"Writing {md_file}.")
+    with open(os.path.join(out_path, md_file), "w", encoding="utf-8") as f:
+        f.write(markdown_str)
+
+
+def _is_module_ignored(module_name: str, ignored_modules: List[str]) -> bool:
+    """Checks if a given module is ignored."""
+    if module_name.split(".")[-1].startswith("_"):
+        return True
+
+    for ignored_module in ignored_modules:
+        if module_name == ignored_module:
+            return True
+
+        # Check is module is subpackage of an ignored package
+        if module_name.startswith(ignored_module + "."):
+            return True
+
+    return False
+
+
+def generate_title(s: str) -> str:
+    """Remove underscores and capitalize first letter to each word"""
+    s = s.replace('_', ' ')
+    s = s.title()
+    return s
+
+
+def create_entity_docs(api_doc_file_dir: Path,
+                       ignored_modules: List[str],
+                       sources_path: Path,
+                       index_file_contents: Optional[List[str]]
+                       ) -> Optional[List[str]]:
+    for item in sources_path.iterdir():
+        if item.name not in ignored_modules:
+            item_name = generate_title(item.stem)
+
+            # Extract zenml import path from sources path
+            # Example: src/zenml/cli/function.py ->  zenml.cli
+            zenml_import_path = '.'.join(item.parts[1:-1])
+
+            module_md = f"# {item_name}\n\n" \
+                        f"::: {zenml_import_path}.{item.stem}\n" \
+                        f"    handler: python\n" \
+                        f"    rendering:\n" \
+                        f"      show_root_heading: true\n" \
+                        f"      show_source: true\n"
+
+            to_md_file(
+                module_md,
+                item.stem,
+                out_path=api_doc_file_dir,
+            )
+
+            if index_file_contents:
+                index_entry = f"# [{item_name}]" \
+                              f"({api_doc_file_dir.name}/{item.stem})\n\n" \
+                              f"::: zenml.{item.stem}\n" \
+                              f"    handler: python\n" \
+                              f"    selection:\n" \
+                              f"        members: false\n"
+
+                index_file_contents.append(index_entry)
+
+    return index_file_contents
+
+
+def create_cli_docs(cli_dev_doc_file_dir: Path,
+                    ignored_modules: List[str],
+                    sources_path: Path,
+                    ) -> None:
+    # TODO [MEDIUM]: Find Solution for issue with click-decorated functions
```

Comment:
One quick comment: The todo parsing is currently only run in the `src` and `tests` directory, so this will not get detected

Reply:
Ah, good to know, I'll create a ticket manually for this

Reply:
unless you just want to add the docs to the todo parser, if you think that's worthwhile I don't see why not, it only includes .py files anyway

Reply:
something worth considering for the future